ceph-proxy/hooks/ceph_hooks.py

```diff
@@ -291,6 +291,11 @@ def client_relation_changed():
             if not is_leader():
                 log("Not leader - ignoring broker request", level=DEBUG)
             else:
+                # client.nova-compute is created when joined
+                # client.nova-compute-ceph-auth-c91ce26f need to be created
+                # in changed hook otherwise the key will not be created
+                ceph.get_named_key(settings["application-name"])
```

Comment:
I think it would be worth noting (in the comment) that this will create the key as a side effect if it's not already present (and we're not actually using the key for anything). 
Also I'm a bit unsure where the name `nova-compute-ceph-auth-c91ce26f` comes from, is it always hardcoded to be this string? Maybe would be good to have an explanation around that too if possible.

Reply:
Thanks for the review, added a bit more info